src/main/java/spoon/support/sniper/internal/IndentationDetector.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * SPDX-License-Identifier: (MIT OR CECILL-C)
+ *
+ * Copyright (C) 2006-2019 INRIA and contributors
+ *
+ * Spoon is available either under the terms of the MIT License (see LICENSE-MIT.txt) of the Cecill-C License (see LICENSE-CECILL-C.txt). You as the user are entitled to choose the terms under which to adopt Spoon.
+ */
+package spoon.support.sniper.internal;
+
+import org.apache.commons.lang3.tuple.Pair;
+import spoon.reflect.declaration.CtCompilationUnit;
+import spoon.reflect.path.CtRole;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Utility class for detecting the indentation style used in a compilation unit.
+ */
+public class IndentationDetector {
+
+	private IndentationDetector() {
+	}
+
+	/**
+	 * Detect the indentation style of the given compilation unit as 1, 2 or 4 spaces or tabs.
+	 *
```

Comment:
Could you add a comment about the strategy that you choosed? (Relying on type members).